src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -243,10 +250,72 @@ public Game getGameFromRandomUser(User user) {
     PageRequest pageRequest = PageRequest.of(pageIndex, 1);
 
     // get the page with the game
-    Page<Game> somePage = gameRepository.getOtherUserWithActiveGameThatWasNotPlayedOn(pageRequest, user.getId(), user,
-        user.getGender(), minAgeDate, maxAgeDate, user.getBlockedUsers(), user.getMatchedUsers());
+    Page<Game> somePage = gameRepository.getOtherUserWithActiveGameThatWasNotPlayedOn(
+      pageRequest, 
+      user.getId(), 
+      user,
+      user.getGender(), 
+      minAgeDate, 
+      maxAgeDate, 
+      user.getBlockedUsers(),
+      user.getMatchedUsers()
+    );
+
+    // subset the page of users to only users that have a haversine distance of less than user.getMaxRange()
+    List<Game> games = somePage.getContent();
+    List<Game> gamesWithDistance = new ArrayList<>();
+    // go over all games that match the other criteria
+    for(Game game : games){
+      // for each game get the user that the game belongs to
+      User gameUser = game.getUser();
+
+      // calculate the distance between the two users
+      double distance = haversineDistance(
+        user.getLatitude(), 
+        user.getLongitude(), 
+        gameUser.getLatitude(), 
+        gameUser.getLongitude()
+      );
+
+      // QUESTION TO REVIEWER: right now i compute the distance for every user i find that matches the other preference criteria. However, I only need to compute it until I find one that has a distance that is 
```

Comment:
I don't understand your question.

Reply:
1. You can always break the loop, especially if  you've found one. So instead of iterating through the whole loop, just immediately return the game if you've found one. 2. If you do so, you raise a 404 error after the loop.

Reply:
right, that would (or should) have been the obvious choice!